### ISSUE TITLE
How to use controlNet with pretrained LoRA?

### ISSUE THREAD

**Issue by zengqixun12** [lllyasviel/ControlNet#205]
I want to combine the pretrained LORA_model and pretrained controlnet_mdoel like "[control_sd15_depth.pth]",and export new model.

How to realize this function?

**Comment by JamesSand** [comment]
I have the same problem with you, is there any solution?

**Comment by zengqixun12** [comment]
> I have the same problem with you, is there any solution?

I've solved this problem, and I combined Lora + Train + ControlNet + change basemodel, export the final result and the model without Stable-Diffusion-WebUI (only use the command line)

**Comment by zengqixun12** [comment]
> I have the same problem with you, is there any solution?

I will release the code in few days ;)

**Comment by JamesSand** [comment]
> > I have the same problem with you, is there any solution?
> 
> I will release the code in few days ;)

thanks a lot. I am also trying to solve this issue. May I ask should we apply lora on the trainable copy of controlnet?

**Comment by zengqixun12** [comment]
> 

Not exactly, Lora  is only effective on the main network and is independent from ControlNet

**Comment by SnowdenLee** [comment]
Hi @zengqixun12, may I ask do you still plan to release the code? Thank you :)

**Comment by gwyong** [comment]
@zengqixun12  Hello, do you still have plan to release the code? Thanks,

**Comment by jiachen0212** [comment]
> > > I have the same problem with you, is there any solution?
> > 
> > 
> > I will release the code in few days ;)
> 
> thanks a lot. I am also trying to solve this issue. May I ask should we apply lora on the trainable copy of controlnet?

You can try: first merge the SD model of base and the parameters of lora, similar to the merger of con parameters and bn parameters. Then train your own data according to the document of train.md. I followed this plan and it worked.
